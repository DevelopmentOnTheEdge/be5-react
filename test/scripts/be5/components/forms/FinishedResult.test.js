import React          from 'react';
import renderer       from 'react-test-renderer';
import FinishedResult from '../../../../../src/scripts/be5/components/forms/FinishedResult';
import {MAIN_DOCUMENT} from "../../../../../src/scripts/be5/constants";


test('snapshot', () => {
  const json = {
    "data":{
      "attributes":{"operationResult": {"status":"finished"}},
      "type":"operationResult",
      "links":{"self":"form/categories/Doc categories/Delete/selectedRows=105"}
    },
    "meta":{"_ts_":"1518764107016"}
  };

  const component = renderer.create(
    <FinishedResult value={json} frontendParams={{documentName: 'test'}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('snapshot custom message', () => {
  const json = {
    "data":{
      "attributes":{"operationResult": {"status":"finished", "message":"Test message."}},
      "type":"operationResult",
      "links":{"self":"form/categories/Doc categories/Delete/selectedRows=105"}
    },
    "meta":{"_ts_":"1518764107016"}
  };

  const component = renderer.create(
    <FinishedResult value={json} frontendParams={{documentName: MAIN_DOCUMENT}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
