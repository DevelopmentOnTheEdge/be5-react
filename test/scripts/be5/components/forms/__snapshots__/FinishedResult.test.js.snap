// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="finishedResult"
>
  <div
    className="mb-3"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Successfully completed.",
      }
    }
  />
</div>
`;

exports[`snapshot custom message 1`] = `
<div
  className="finishedResult"
>
  <div
    className="mb-3"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test message.",
      }
    }
  />
  <button
    className="btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
</div>
`;
